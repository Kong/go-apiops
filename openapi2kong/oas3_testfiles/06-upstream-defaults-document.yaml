# x-kong-upstream-defaults can be specified and will be honored on each level

openapi: '3.0.0'
info:
  title: Simple API overview
  version: v2
servers:
  - url: https://server1.com/
x-kong-upstream-defaults:
  slots: 1000

paths:
  /path1:
    # this should create routes attached to the document-level service
    get:
      operationId: uses-doc-service
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
  /path2:
    # specify upstream defaults to force another upstream+service being
    # created at path level, should inherit server block from the document level
    x-kong-upstream-defaults:
      slots: 2000
    get:
      operationId: uses-path-service
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
    post:
      # specify upstream defaults to force another upstream+service being
      # created at operation level, should inherit server block from the document level
      x-kong-upstream-defaults:
        slots: 3000
      operationId: uses-ops-service
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
