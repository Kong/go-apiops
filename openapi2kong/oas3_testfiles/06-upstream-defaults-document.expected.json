{
  "_format_version": "3.0",
  "services": [
    {
      "host": "simple-api-overview.upstream",
      "id": "0907c4ab-d9e4-5d21-813b-c57a97eeaad9",
      "name": "simple-api-overview",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "663104d8-7e60-525d-b506-e42971b4466b",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_uses-doc-service",
          "paths": [
            "~/path1$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_06-upstream-defaults-document.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_06-upstream-defaults-document.yaml"
      ]
    },
    {
      "host": "simple-api-overview_path2.upstream",
      "id": "a79c5a8c-0924-599e-9412-39f5a4ff0c3e",
      "name": "simple-api-overview_path2",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "c1b84366-8ff8-57b9-b118-bedd1b9ab1c8",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_uses-path-service",
          "paths": [
            "~/path2$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_06-upstream-defaults-document.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_06-upstream-defaults-document.yaml"
      ]
    },
    {
      "host": "simple-api-overview_uses-ops-service.upstream",
      "id": "5d05e040-c551-50a1-90e7-6af5d3a7c0dc",
      "name": "simple-api-overview_uses-ops-service",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "ca1c6faa-7076-5a09-9346-9c9dc3e019cb",
          "methods": [
            "POST"
          ],
          "name": "simple-api-overview_uses-ops-service",
          "paths": [
            "~/path2$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_06-upstream-defaults-document.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_06-upstream-defaults-document.yaml"
      ]
    }
  ],
  "upstreams": [
    {
      "id": "811c42d6-ef18-5296-a550-7dca2262b4d8",
      "name": "simple-api-overview.upstream",
      "slots": 1000,
      "tags": [
        "OAS3_import",
        "OAS3file_06-upstream-defaults-document.yaml"
      ],
      "targets": [
        {
          "tags": [
            "OAS3_import",
            "OAS3file_06-upstream-defaults-document.yaml"
          ],
          "target": "server1.com:443"
        }
      ]
    },
    {
      "id": "ef3215c6-42e1-5380-ad2d-24f3b2d05972",
      "name": "simple-api-overview_path2.upstream",
      "slots": 2000,
      "tags": [
        "OAS3_import",
        "OAS3file_06-upstream-defaults-document.yaml"
      ],
      "targets": [
        {
          "tags": [
            "OAS3_import",
            "OAS3file_06-upstream-defaults-document.yaml"
          ],
          "target": "server1.com:443"
        }
      ]
    },
    {
      "id": "e2aac4d8-a96f-50ec-b71e-980e6e581a50",
      "name": "simple-api-overview_uses-ops-service.upstream",
      "slots": 3000,
      "tags": [
        "OAS3_import",
        "OAS3file_06-upstream-defaults-document.yaml"
      ],
      "targets": [
        {
          "tags": [
            "OAS3_import",
            "OAS3file_06-upstream-defaults-document.yaml"
          ],
          "target": "server1.com:443"
        }
      ]
    }
  ]
}